### ISSUE TITLE
How to add an example program in Visual Studio and how to adaptor NetFlow layer to all OS platform?

### ISSUE THREAD

**Issue by angelyouyou** [seladb/PcapPlusPlus#752]
**Question1:**
How to generate the vs file below on windows.I write these two files manually, but I am worried that this will produce errors (such as GUID, etc

![image](https://user-images.githubusercontent.com/12936259/140285799-bc979d8e-759a-4514-b271-bb78066f1cc9.png)

![image](https://user-images.githubusercontent.com/12936259/140285955-d46cf62b-ae9a-456b-91e3-c5ec8ca2664d.png)

**Question2:**
How  to adaptor NetFlow layer to all OS platform? I test it only on MacOS platform and add some fiels files.
Are there any related fiels that need to be modified or added?
![image](https://user-images.githubusercontent.com/12936259/140286813-4a83ecbe-4773-4122-ba62-dd04ca8809ce.png)

**Comment by seladb** [comment]
> **Question1:**
> How to generate the vs file below on windows.I write these two files manually, but I am worried that this will produce errors (such as GUID, etc

I think the safest way would be to take an existing example project file (for example: `Arping.vcxproj.template` and `Arping.vcxproj.filters`) and rename the necessary fields in them. The best way to know what you need to rename is to compare files from different projects, for example: compare `Arping.vcxproj.template` with `DNSResolver.vcxproj.template`, and compare `Arping.vcxproj.filters` with `DNSResolver.vcxproj.filters` (assuming you only have a `main.cpp` file in the example). Then you can generate new GUIDs.

Another thing to remember is to add this new projects to `PcapPlusPlus-Examples.sln`



> **Question2:**
> How to adaptor NetFlow layer to all OS platform? I test it only on MacOS platform and add some fiels files.
> Are there any related fiels that need to be modified or added?

I'm not sure I understand the question. Are you asking how to add a new layer and make sure it's built on all platforms?
Basically once you add your new layer to `Packet++/header` and `Packet++/src` this layer will automatically be built on all platforms except Windows / Visual Studio. For Visual Studio you also need to add these new files to `Packet++.vcxproj.template` and `Packet++.vcxproj.filters`. When you open these files you can see how other layers were added and add your new files in the same way

**Comment by angelyouyou** [comment]
> Packet++.vcxproj.filters



> > **Question1:**
> > How to generate the vs file below on windows.I write these two files manually, but I am worried that this will produce errors (such as GUID, etc
> 
> I think the safest way would be to take an existing example project file (for example: `Arping.vcxproj.template` and `Arping.vcxproj.filters`) and rename the necessary fields in them. The best way to know what you need to rename is to compare files from different projects, for example: compare `Arping.vcxproj.template` with `DNSResolver.vcxproj.template`, and compare `Arping.vcxproj.filters` with `DNSResolver.vcxproj.filters` (assuming you only have a `main.cpp` file in the example). Then you can generate new GUIDs.
> 
> Another thing to remember is to add this new projects to `PcapPlusPlus-Examples.sln`
> 
> > **Question2:**
> > How to adaptor NetFlow layer to all OS platform? I test it only on MacOS platform and add some fiels files.
> > Are there any related fiels that need to be modified or added?
> 
> I'm not sure I understand the question. Are you asking how to add a new layer and make sure it's built on all platforms? Basically once you add your new layer to `Packet++/header` and `Packet++/src` this layer will automatically be built on all platforms except Windows / Visual Studio. For Visual Studio you also need to add these new files to `Packet++.vcxproj.template` and `Packet++.vcxproj.filters`. When you open these files you can see how other layers were added and add your new files in the same way

@seladb
Thanks. I have add these files to my local project.I will test it.
